### ISSUE TITLE
Build ninja from source if prebuilt binary is absent for a platform

### ISSUE THREAD

**Issue by dmkurilov** [bazelbuild/rules_foreign_cc#1019]
## The problem

ninja's prebuilt binaries are available for [a limited number of platforms](https://github.com/bazelbuild/rules_foreign_cc/blob/2c6262f8f487cd3481db27e2c509d9e6d30bfe53/toolchains/prebuilt_toolchains.py#L103-L116).

For example, if you pick a linux_aarch64, you'll get an error:
```bah
No matching toolchains found for types @rules_foreign_cc~0.9.0//toolchains:ninja_toolchain
```

## How to reproduce

Please, take a look at https://github.com/bazelbuild/rules_foreign_cc/issues/1018, section "How to reproduce".

## Expected behavour

ninja is built from source if there is no binary for a platform.

**Comment by jsharpe** [comment]
I think this is just that the bzlmod MODULE.bazel file doesn't register the built ninja toolchain by default. You should be able to register the toolchain in your root MODULE.bazel file to work around this.

**Comment by tkekan** [comment]
@jsharpe 
Sorry can you please provide some details on how to work around this ? I'm facing similar issue wherein a third-party ( libevent) which is a transitive dependency is being built using the cmake rule. But this complains about 
unable to find toolchain since I'm building this on OL8 arm64 docker running on Apple silicon. 

`
third-party/rules_libevent/BUILD:26:6: While resolving toolchains for target //third-party/rules_libevent:event (47d3177): No matching toolchains found for types @@rules_foreign_cc~//toolchains:ninja_toolchain.
`
